Enterprise-CMCS/macpro-mdct-seds
# Summary tab

Satisfies: [#3839](https://qmacbis.atlassian.net/browse/OY2-3839?atlOrigin=eyJpIjoiZjBhYzNjMzkwMTg2NDc4Yzk1NTMyODU4MTdmYWZhNzUiLCJwIjoiaiJ9)

This PR adds functionality for the Summary Tab, which is available on every form page. The summary tab adds together all previous tabs (Ages 0-1, 1-3, etc) and presents them in an uneditable format for review by the user. 

**package.json**
Add babel-plugin-transform-inline-environment-variables: Allows Jest to read environment variables

Add jsonpath: Used for extracting data from json files 

**services/ui-src/package.json**
Add redux-mock-store: Allows mocking of provider store to be used by Jest

**services/ui-src/src/components/SummaryTab/SummaryTab.js**
Main functionality: Calculates total of inputs from all age range tabs and displays each question with totals for user review. 

The addition is done by looping through all questions from redux, matching the question ID, and doing addition. A "new" question is created by pulling default values from a matching question and present in the Summary Tab.

Also, moved this into a sub-directory for organizational reasons. The new directory houses both SummaryTab.js and SummaryTab.test.js. No scss file is currently needed for this functionality.

**services/ui-src/src/providerMocks/currentFormMock_21E.js**
Dummy data in the correct format to be used by multiple Jest tests.

**services/ui-src/src/components/SummaryTab/SummaryTab.test.js**
Tests to check functionality:

- Check that the tab is rendering
- Check that the header is correct
- Check that girdwithtotal headers are correct (both top and left-side headers)
- Check that the math is correct in the first row of cells in the first gridwithtotal
